### PR TITLE
Feature/boas 464 create folders

### PR DESCRIPTION
**Pull request by Xfarya**
## Describe your changes
create folders

## Issue ticket number and link
BOAS-464 (part of 430) https://pins-ds.atlassian.net/browse/BOAS-464

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/repositories/case.repository.js [hunk 1]
```diff
@@ -411,11 +411,13 @@ export const createNewStatuses = (id, status) => {
...
  * @param {number} id
  * @param {{status: string | object, data: {regionNames?: string[]}, currentStatuses: object[], setReference: boolean}} updateData
+ * @param {Promise<any>[]} createFolders
  * @returns {any}
  */
 export const updateApplicationStatusAndDataById = (
 	id,
-	{ status, data, currentStatuses, setReference = false }
+	{ status, data, currentStatuses, setReference = false },
+	/** @type {any[] | Promise<any>} */ createFolders
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/case.repository.js:420]: Could we please replace `any` with the accurate data types?

File: apps/api/src/server/repositories/folder.repository.js [hunk 2]
```diff
@@ -8,3 +8,65 @@ import { databaseConnector } from '../utils/database-connector.js';
 export const getByCaseId = (caseId) => {
 	return databaseConnector.folder.findMany({ where: { caseId } });
 };
+
+export const createFolders = () => {
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/folder.repository.js:12]: Could we please add the JSDoc to this function?

File: apps/api/src/server/repositories/case.repository.js (hunk 1 above)
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/case.repository.js:420]: I would not call this `createFolders`, but something a bit more generic. We could in theory pass any other transaction that we might want here.

File: apps/api/src/server/repositories/case.repository.js [hunk 3]
```diff
@@ -439,6 +441,8 @@ export const updateApplicationStatusAndDataById = (
 		transactions.push(assignApplicationReference(id));
 	}
 
+	transactions.push(createFolders);
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/case.repository.js:444]: How does `push` work? If `createFolders` is an array, is it merged into `transactions` as a 1D array?

File: apps/api/src/server/repositories/folder.repository.js [hunk 4]
```diff
@@ -8,3 +8,65 @@ import { databaseConnector } from '../utils/database-connector.js';
...
+	return databaseConnector.folder.create({
+		data: [
+			{
+				displayNameEn: 'Project management',
+				displayOrder: 100,
+				type: 'folder'
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/folder.repository.js:18]: I don't believe we have the column `type` in the `folder` table?

File: apps/api/src/server/repositories/folder.repository.js [hunk 5]
```diff
@@ -8,3 +8,65 @@ import { databaseConnector } from '../utils/database-connector.js';
 export const getByCaseId = (caseId) => {
 	return databaseConnector.folder.findMany({ where: { caseId } });
 };
+
+export const createFolders = () => {
+	return databaseConnector.folder.create({
+		data: [
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/folder.repository.js:14]: Could we break out this array into a variable elsewhere in this file? This is for readability purposes

**Review by mayya-bond-kc — CHANGES_REQUESTED** [review]
Could we add some tests for this as well please?

File: apps/api/src/server/repositories/folder.repository.js [hunk 6]
```diff
@@ -8,3 +8,25 @@ import { databaseConnector } from '../utils/database-connector.js';
 export const getByCaseId = (caseId) => {
 	return databaseConnector.folder.findMany({ where: { caseId } });
 };
+
+export const createFolders = (/** @type {number} */ caseId) => {
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/folder.repository.js:12]: Could we please add JSDocs to this function?

File: apps/api/src/server/repositories/folder.repository.js [hunk 7]
```diff
@@ -8,3 +8,25 @@ import { databaseConnector } from '../utils/database-connector.js';
...
+	const createdFolders = {
+		data: folderHold.map((folder) => ({ ...folder, caseId }))
+	};
```
Comment (mayya-bond-kc) [inline apps/api/src/server/repositories/folder.repository.js:15]: Nice 👍

File: apps/api/src/server/applications/application/application.service.js [hunk 8]
```diff
@@ -95,12 +95,16 @@ export const startApplication = async (id) => {
...
-	});
+	await caseRepository.updateApplicationStatusAndDataById(
+		caseDetails.id,
+		{
+			status: nextStatusForRepository,
+			data: {},
+			currentStatuses: caseDetails.CaseStatus,
+			setReference: true
+		},
+		[folderRepository.createFolders()]
```
Comment (mayya-bond-kc) [inline apps/api/src/server/applications/application/application.service.js:106]: We need to add the case ID to the arguments of this function, right?

**Review by mayya-bond-kc — CHANGES_REQUESTED** [review]
As per my previous comment, could we please add tests for this?

File: apps/api/src/server/applications/application/application.service.js (hunk 8 above)
Comment (Xfarya) [reply apps/api/src/server/applications/application/application.service.js:106]: yes i thought i did hmm

Comment (mayya-bond-kc) [reply apps/api/src/server/applications/application/application.service.js:105]: This is still missing

File: apps/api/src/server/applications/application/application.service.js [hunk 9]
```diff
@@ -5,7 +5,6 @@ import { breakUpCompoundStatus } from '../../utils/break-up-compound-status.js';
 import { buildAppealCompundStatus } from '../../utils/build-appeal-compound-status.js';
 import { mapApplicationDetails } from '../../utils/mapping/map-case-details.js';
 import { mapFolderDetails } from '../../utils/mapping/map-folder-details.js';
-// import { sendMessage } from '../../utils/service-bus-sender.js';
```
Comment (mayya-bond-kc) [inline apps/api/src/server/applications/application/application.service.js:8]: Could we leave this in please? This is a placeholder for when we will be sending messages to the message queue

File: apps/api/src/server/applications/application/application.service.js (hunk 8 above)
Comment (Xfarya) [reply apps/api/src/server/applications/application/application.service.js]: it's showing up as there on my github - has it maybe updated?

File: apps/api/src/server/applications/application/application.service.js (hunk 9 above)
Comment (Xfarya) [reply apps/api/src/server/applications/application/application.service.js]: i didn't remove this

**Review by mayya-bond-kc — APPROVED** [review]
LGTM

**Pull request merged**